File: docs/tutorials/llm/openai.py
# openai_nosana.py

from openai import OpenAI

client = OpenAI(
    base_url="https://<nosana-job-id>.node.k8s.prd.nos.ci/v1",
)

# Use the following func to get the available models
# model_list = client.models.list()
# print(model_list)

chat_completion = client.chat.completions.create(
    model="R1-Qwen-1.5B",
    messages=[
        {
            "role": "user",
            "content": "Explain superconductors like I'm five years old"
        }
    ],
    stream=True,
)

for chunk in chat_completion:
    print(chunk.choices[0].delta.content or "", end="")

File: docs/tutorials/llm/openai.ts
// openai.js
import OpenAI from "openai";
const openai = new OpenAI({
  baseURL: "https://<nosana-job-id>.node.k8s.prd.nos.ci/v1"
});

const models = await openai.models.list();
console.log(models.data);

const completion = await openai.chat.completions.create({
  model: "R1-Qwen-1.5B",
  messages: [
    {
      role: "user",
      content: "Write a haiku about recursion in programming.",
    },
  ],
});

console.log(completion.choices[0].message.content);
